gormschema/gorm.go

```diff
@@ -13,55 +14,80 @@ import (
 )
 
 // New returns a new Loader.
-func New(dialect string) *Loader {
-	return &Loader{dialect: dialect}
+func New(dialect string, config ...func(*gorm.Config)) *Loader {
+	gormConfig := &gorm.Config{}
+
+	for _, c := range config {
+		c(gormConfig)
+	}
+
+	return &Loader{
+		dialect: dialect,
+		config:  gormConfig,
+	}
 }
 
 // Loader is a Loader for gorm schema.
 type Loader struct {
 	dialect string
+	config  *gorm.Config
+}
+
+func (l *Loader) Load(models ...interface{}) (string, error) {
+	di, err := l.getDialector()
+	if err != nil {
+		return "", err
+	}
+
+	db, err := gorm.Open(di, l.config)
+	if err != nil {
+		return "", err
+	}
+
+	if err := db.AutoMigrate(models...); err != nil {
+		return "", err
+	}
+
+	s, ok := recordriver.Session("gorm")
+	if !ok {
+		return "", errors.New("failed to retrieve recordriver session")
+	}
+	return s.Stmts(), nil
+}
+
+func WithForeignKeyConstraintDisabled() func(*gorm.Config) {
```

Comment:
```suggestion
func WithForeignKeys(b bool) func(*gorm.Config) {
```

I would prefer a shorter API like this.

Reply:
updated